docs/extend/static-code-analysis.md

```diff
@@ -0,0 +1,86 @@
+# Static Code Analysis
+
+Static code analysis is the process of analyzing the source code against a set of rules to find bugs, code smells, and security vulnerabilities. This is a great way to improve the quality of your code and to find potential issues before they are deployed to production. An example is validating the typings of a function to ensure that the function is called with the correct arguments.
+
+Flarum provides a static code analysis package based on [PHPStan](https://phpstan.org/) that can be added to your extension. In this guide, we will show you how to add the package to your extension and how to run the analysis.
+
+## Setup
+
+:::tip [Flarum CLI](https://github.com/flarum/cli)
+
+You can use the CLI to automatically add and update the infrastructure for phpstan to your code:
+
+```bash
+$ flarum-cli infra phpstan
+```
+
+:::
+
+First you need to require the `flarum/phpstan` package in your extension. You can do this by running the following command in the root of our extension:
+
+```bash
+composer require --dev flarum/phpstan:^1.0
+```
+
+Next, you need to create a `phpstan.neon` file in the root of your extension. This file contains [the configuration for PHPStan](https://phpstan.org/config-reference). You can copy the following configuration into the file:
+
+```neon
+includes:
+  - vendor/php-packages/phpstan/extension.neon
+
+parameters:
+	# The level will be increased in Flarum 2.0
+	level: 5
+	paths:
+	  - src
+	  - extend.php
+	excludePaths:
+		- *.blade.php
+	checkMissingIterableValueType: false
+	databaseMigrationsPath: ['framework/core/migrations']
```

Comment:
This path also seems strange for out-of-tree extensions.

Reply:
good catch!
```suggestion
	databaseMigrationsPath: ['migrations']
```